apps/studio/src/lib/db/clients/mysql.ts

```
@@ -1445,6 +1451,321 @@ export class MysqlClient extends BasicDatabaseClient<ResultType> {
+              const schema = schemaRow['SCHEMA_NAME'];
+              schemaGrantsMap[schema] = schemaGrantsMap[schema] || {};
+              privileges.forEach(priv => {
+                schemaGrantsMap[schema][priv] = true;
+              });
+            }
+          }
+        }
+      }
+    }
+  
+    const { rows: schemaRows } = await this.driverExecuteSingle(
+      `SELECT schema_name
+       FROM information_schema.schemata
+       WHERE schema_name NOT IN ('information_schema', 'performance_schema', 'sys')`,
+      { connection }
+    );
+  
+    for (const row of schemaRows) {
+      const schema = row['SCHEMA_NAME'];
+      if (!schemaGrantsMap[schema]) {
+        schemaGrantsMap[schema] = {};
+        allPrivileges.forEach(priv => {
+          schemaGrantsMap[schema][priv] = false;
+        });
+      }
+    }
+  
+    const userPrivileges: UserPrivileges[] = Object.entries(schemaGrantsMap).map(([schema, privileges]) => ({
+      schema: schema,
+
+      select: !!privileges["SELECT"],
+      insert: !!privileges["INSERT"],
+      update: !!privileges["UPDATE"],
+      delete: !!privileges["DELETE"],
+      execute: !!privileges["EXECUTE"],
+      show_View: !!privileges["SHOW VIEW"],
+      create: !!privileges["CREATE"],
+      alter: !!privileges["ALTER"],
+      references: !!privileges["REFERENCES"],
+      index: !!privileges["INDEX"],
+      create_View: !!privileges["CREATE VIEW"],
+      create_Routine: !!privileges["CREATE ROUTINE"],
+      alter_Routine: !!privileges["ALTER ROUTINE"],
+      drop: !!privileges["DROP"],
+      event: !!privileges["EVENT"],
+      trigger: !!privileges["TRIGGER"],
+      grant_Option: !!privileges["GRANT OPTION"],
+      lock_Tables: !!privileges["LOCK TABLES"],
+      create_Temporary_Tables: !!privileges["CREATE TEMPORARY TABLES"],
+    }));
+  
+    return userPrivileges;
+  }
+  
+  async applyUserChanges(changes: any[]): Promise<{ success: boolean; error?: string }> {
```

The changes should be typed here with the model that you made earlier. I also still don't think that this needs to return anything, you can just throw any errors that come up and catch them in the frontend.